scripts/reorg.fish

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env fish
+
+mv service-tests/runner lib/service-test-runner
+mv service-tests/helpers/response-fixtures.js lib/response-fixtures.js
+mv service-tests/helpers/validators.js lib/service-test-validators.js
+
+mv service-tests/README.md doc/service-tests.md
+
+mkdir -p services
+mv service-tests/.eslintrc.yml services/
+
+mv services/appveyor.js services/appveyor/appveyor.js
+
+for file in service-tests/*.js
+  set service (string match -r "^service-tests/(.*).js\$" $file)[2]
+  mkdir services/$service
+  mv $file services/$service/$service.tester.js
+end
+
+mv service-tests/helpers/nuget-fixtures.js services/nuget/nuget-fixtures.js
```

Comment:
I'm not sure I have an obviously better suggestion here, but having stuff like this and `test-validators.js` in the root of the services dir feels like maybe conceptually the wrong place to put them.. I get that making a subdir for them then makes that subdir behave like a 'service', which is also wrong. Maybe there is a case for `/lib/service` or `/lib/testing` ?

The correct answer to that may only really become clear once we start feeling our way around  some of the issues we're discussing in https://github.com/badges/shields/issues/1358#issuecomment-373253805

Reply:
`lib/testing` seems like a great name. `lib/service` seems a little ambiguous. Does it contain service-related helpers (perhaps including `base.js`) or helpers for particular types of external services? Maybe `lib/service-specific` would be clearer.

Yea… `service/lib` or `service/common` isn't quite right either. It might invite people to misclassify helpers there which ought to be placed in `lib/`.

Though… huh. Maybe that's just backward thinking. Maybe _all_ the helper functions designed to be used from the services should be moved from `lib/` into `services/common`. Or `services/COMMON` which sorts at the top. They certainly would be easier to find. I'm thinking of the text formatters, error helpers, and the version and license helpers. Honestly that's most of what's in `lib/`. What's left would basically be "core lib".

Thoughts?